### ISSUE TITLE
Tab syncing failed: "Detected creation of folder loop:"

### ISSUE THREAD

**Issue by srd424** [floccusaddon/floccus#1107]
### Which version of floccus are you using?

4.12.0

### Sync method

{"label"=>"WebDAV"}

### Which browser are you using?

Chrome 99.0.4844.84

### What kind of WebDAV server are you using? (if relevant)

https://github.com/hacdias/webdav 4.1.1

### Describe the Bug

Error while trying to sync Chrome tabs to webdav server:

"Syncing failed with Detected creation of folder loop: Moving 32 to 32, but it already contains the new parent node"

I currently have two Chrome windows open each containing a number of tabs, I can't see anything particularly odd about any of them :) I am using the "overwrite cloud with local" sync option.

### Expected Behavior

Expected tabs to sync to server with no error.

### To Reproduce

Can't help with that unless it recurs, but I will upload a debug log.

### Debug log provided

- [X] I have provided a debug log file

**Comment by srd424** [comment]
Closing the last tab in the second window seems to have allowed the sync to succeed, if that helps?

**Comment by srd424** [comment]
Ah, just opening a second window looks like it may be enough to cause the problem?

**Comment by marcelklehr** [comment]
I'll look into this. Thanks for reporting!

**Comment by marcelklehr** [comment]
@srd424 Do you still have this problem with the latest version?

**Comment by srd424** [comment]
Yes, unfortunately it looks like I still do :( I have uploaded another debug log - floccus-4.15.0-2022-06-12.log - let me know if you need the unredacted version.

**Comment by marcelklehr** [comment]
@srd424 The log file you provided is from a different sync run, please make sure to generate the log file immediately after the error occurs, as the logs will be overridden by the latest sync run

**Comment by srd424** [comment]
I did it immediately after testing, so not sure what happened. I will look
again tomorrow

**Comment by srd424** [comment]
OK, I double checked the file contents this time: floccus-4.15.0-2022-06-13 (1).log

It only seems to happen if a second window is open, and even then not all the time. Possibly dependent on whether there is more than one tab open, whether a blank/start tab is showing, and what website(s) are in the tabs in the second window, but I can't discern a definite pattern.

**Comment by marcelklehr** [comment]
Your tree on the server seems wrong. You have the root folder, then an untitled folder and in there should only be bookmarks, but you have another folder in there. Not sure why that happens. As a short-term fix you can move that folder out of the other folder.

**Comment by srd424** [comment]
That untitled folder is created by floccus for some reason - I've just tried deleting the file on the server and repushing the tabs, but it gets recreated. Is it something do with whatever Chrome API you are using to enumerate the tabs? Does it represent the window the tabs are in or something?

**Comment by marcelklehr** [comment]
The untitled folder is fine, but it seems to contain another folder, which it shouldn't.

**Comment by srd424** [comment]
Oh, I see. Floccus is creating it like that - I can reproduce the error in the bug title even after nuking the file on the server and repushing. If I open a new window and open a website in the new tab, floccus seems to create two untitled top level folders, one for each window, presumably as expected. It seems that maybe if only the "new tab/start page" tab is open in the second window, then Floccus creates the folder for the second window nested below the first?

Even with a website open in the second window, the bookmark for that tab gets put in the wrong folder...

**Comment by marcelklehr** [comment]
> Even with a website open in the second window, the bookmark for that tab gets put in the wrong folder...

I was able to reproduce this now. I will try to come up with a fix :)
